diff --git a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/1/.hash b/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/1/.hash
deleted file mode 100644
index efb4666..0000000
--- a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/1/.hash
+++ /dev/null
@@ -1 +0,0 @@
-467acd9434d0afec502dd4bb33140f0d98025d29
diff --git a/aidl/mtkpower/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl b/aidl/mtkpower/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
deleted file mode 100644
index 91eafd7..0000000
--- a/aidl/mtkpower/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
+++ /dev/null
@@ -1,15 +0,0 @@
-/*
- * Copyright (C) 2024 The LineageOS Project
- *
- * SPDX-License-Identifier: Apache-2.0
- */
-
-package vendor.mediatek.hardware.mtkpower;
-
-@VintfStability
-interface IMtkPowerCallback {
-    oneway void mtkPowerHint(in int hint, in int duration);
-    oneway void notifyAppState(in String pack, in String act, in int pid, in int state,
-        in int uid);
-    oneway void notifyScnUpdate(in int hint, in int data);
-}
diff --git a/aidl/mtkpower/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl b/aidl/mtkpower/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
deleted file mode 100644
index ffcf3d3..0000000
--- a/aidl/mtkpower/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
+++ /dev/null
@@ -1,22 +0,0 @@
-/*
- * Copyright (C) 2024 The LineageOS Project
- *
- * SPDX-License-Identifier: Apache-2.0
- */
-
-package vendor.mediatek.hardware.mtkpower;
-
-import vendor.mediatek.hardware.mtkpower.IMtkPowerCallback;
-
-@VintfStability
-interface IMtkPowerService {
-    oneway void mtkCusPowerHint(in int hint, in int data);
-    oneway void mtkPowerHint(in int hint, in int data);
-    oneway void notifyAppState(in String pack, in String act, in int pid, in int state,
-        in int uid);
-    int querySysInfo(in int cmd, in int param);
-    int setMtkPowerCallback(in IMtkPowerCallback callback);
-    int setMtkScnUpdateCallback(in int hint, in IMtkPowerCallback callback);
-    int setSysInfo(in int type, in String data);
-    void setSysInfoAsync(in int type, in String data);
-}
diff --git a/aidl/mtkpower/Android.bp b/interfaces/hardware/mtkpower/aidl/Android.bp
similarity index 78%
rename from aidl/mtkpower/Android.bp
rename to interfaces/hardware/mtkpower/aidl/Android.bp
index d34ef83..6683b88 100644
--- a/aidl/mtkpower/Android.bp
+++ b/interfaces/hardware/mtkpower/aidl/Android.bp
@@ -1,11 +1,11 @@
-/*
- * Copyright (C) 2024 The LineageOS Project
- *
- * SPDX-License-Identifier: Apache-2.0
- */
+//
+// SPDX-FileCopyrightText: The LineageOS Project
+// SPDX-License-Identifier: Apache-2.0
+//
 
 aidl_interface {
     name: "vendor.mediatek.hardware.mtkpower",
+    system_ext_specific: true,
     vendor_available: true,
     srcs: ["vendor/mediatek/hardware/mtkpower/*.aidl"],
     stability: "vintf",
diff --git a/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/1/.hash b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/1/.hash
new file mode 100644
index 0000000..2a29e97
--- /dev/null
+++ b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/1/.hash
@@ -0,0 +1 @@
+d8705e0f9df6e836a52603c5b505c6be92265c71
diff --git a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
similarity index 78%
rename from aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
rename to interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
index 9b3c26a..d115385 100644
--- a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
+++ b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
@@ -1,8 +1,7 @@
-/*
- * Copyright (C) 2024 The LineageOS Project
- *
- * SPDX-License-Identifier: Apache-2.0
- */
+//
+// SPDX-FileCopyrightText: 2026 The LineageOS Project
+// SPDX-License-Identifier: Apache-2.0
+//
 ///////////////////////////////////////////////////////////////////////////////
 // THIS FILE IS IMMUTABLE. DO NOT EDIT IN ANY CASE.                          //
 ///////////////////////////////////////////////////////////////////////////////
@@ -24,7 +23,7 @@
 package vendor.mediatek.hardware.mtkpower;
 @VintfStability
 interface IMtkPowerCallback {
-  oneway void mtkPowerHint(in int hint, in int duration);
-  oneway void notifyAppState(in String pack, in String act, in int pid, in int state, in int uid);
-  oneway void notifyScnUpdate(in int hint, in int data);
+  oneway void mtkPowerHint(in int hint, in int data);
+  oneway void notifyAppState(in String packageName, in String activityName, in int pid, in int state, in int uid);
+  oneway void notifyScnUpdate(in int scn, in int data);
 }
diff --git a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
similarity index 70%
rename from aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
rename to interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
index bb52679..1739081 100644
--- a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
+++ b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
@@ -1,8 +1,7 @@
-/*
- * Copyright (C) 2024 The LineageOS Project
- *
- * SPDX-License-Identifier: Apache-2.0
- */
+//
+// SPDX-FileCopyrightText: 2026 The LineageOS Project
+// SPDX-License-Identifier: Apache-2.0
+//
 ///////////////////////////////////////////////////////////////////////////////
 // THIS FILE IS IMMUTABLE. DO NOT EDIT IN ANY CASE.                          //
 ///////////////////////////////////////////////////////////////////////////////
@@ -24,12 +23,15 @@
 package vendor.mediatek.hardware.mtkpower;
 @VintfStability
 interface IMtkPowerService {
+  int perfCusLockHint(in int hint, in int duration, in int pid);
+  int perfLockAcquire(in int hdl, in int duration, in int[] boostList, in int pid, in int reserved);
+  oneway void perfLockRelease(in int hdl, in int reserved);
+  int perfLockReleaseSync(in int hdl, in int reserved);
   oneway void mtkCusPowerHint(in int hint, in int data);
   oneway void mtkPowerHint(in int hint, in int data);
-  oneway void notifyAppState(in String pack, in String act, in int pid, in int state, in int uid);
   int querySysInfo(in int cmd, in int param);
   int setMtkPowerCallback(in vendor.mediatek.hardware.mtkpower.IMtkPowerCallback callback);
-  int setMtkScnUpdateCallback(in int hint, in vendor.mediatek.hardware.mtkpower.IMtkPowerCallback callback);
-  int setSysInfo(in int type, in String data);
-  void setSysInfoAsync(in int type, in String data);
+  int setMtkScnUpdateCallback(in int scn, in vendor.mediatek.hardware.mtkpower.IMtkPowerCallback callback);
+  int setSysInfo(in int cmd, in String data);
+  oneway void setSysInfoAsync(in int cmd, in String data);
 }
diff --git a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
similarity index 78%
rename from aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
rename to interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
index f5dd848..d115385 100644
--- a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
+++ b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
@@ -1,8 +1,7 @@
-/*
- * Copyright (C) 2024 The LineageOS Project
- *
- * SPDX-License-Identifier: Apache-2.0
- */
+//
+// SPDX-FileCopyrightText: 2026 The LineageOS Project
+// SPDX-License-Identifier: Apache-2.0
+//
 ///////////////////////////////////////////////////////////////////////////////
 // THIS FILE IS IMMUTABLE. DO NOT EDIT IN ANY CASE.                          //
 ///////////////////////////////////////////////////////////////////////////////
@@ -24,8 +23,7 @@
 package vendor.mediatek.hardware.mtkpower;
 @VintfStability
 interface IMtkPowerCallback {
-    oneway void mtkPowerHint(in int hint, in int duration);
-    oneway void notifyAppState(in String pack, in String act, in int pid, in int state,
-        in int uid);
-    oneway void notifyScnUpdate(in int hint, in int data);
+  oneway void mtkPowerHint(in int hint, in int data);
+  oneway void notifyAppState(in String packageName, in String activityName, in int pid, in int state, in int uid);
+  oneway void notifyScnUpdate(in int scn, in int data);
 }
diff --git a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
similarity index 58%
rename from aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
rename to interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
index 016bc0c..1739081 100644
--- a/aidl/mtkpower/aidl_api/vendor.mediatek.hardware.mtkpower/1/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
+++ b/interfaces/hardware/mtkpower/aidl/aidl_api/vendor.mediatek.hardware.mtkpower/current/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
@@ -1,8 +1,7 @@
-/*
- * Copyright (C) 2024 The LineageOS Project
- *
- * SPDX-License-Identifier: Apache-2.0
- */
+//
+// SPDX-FileCopyrightText: 2026 The LineageOS Project
+// SPDX-License-Identifier: Apache-2.0
+//
 ///////////////////////////////////////////////////////////////////////////////
 // THIS FILE IS IMMUTABLE. DO NOT EDIT IN ANY CASE.                          //
 ///////////////////////////////////////////////////////////////////////////////
@@ -22,16 +21,17 @@
 // later when a module using the interface is updated, e.g., Mainline modules.
 
 package vendor.mediatek.hardware.mtkpower;
-import vendor.mediatek.hardware.mtkpower.IMtkPowerCallback;
 @VintfStability
 interface IMtkPowerService {
-    oneway void mtkCusPowerHint(in int hint, in int data);
-    oneway void mtkPowerHint(in int hint, in int data);
-    oneway void notifyAppState(in String pack, in String act, in int pid, in int state,
-        in int uid);
-    int querySysInfo(in int cmd, in int param);
-    int setMtkPowerCallback(in IMtkPowerCallback callback);
-    int setMtkScnUpdateCallback(in int hint, in IMtkPowerCallback callback);
-    int setSysInfo(in int type, in String data);
-    void setSysInfoAsync(in int type, in String data);
+  int perfCusLockHint(in int hint, in int duration, in int pid);
+  int perfLockAcquire(in int hdl, in int duration, in int[] boostList, in int pid, in int reserved);
+  oneway void perfLockRelease(in int hdl, in int reserved);
+  int perfLockReleaseSync(in int hdl, in int reserved);
+  oneway void mtkCusPowerHint(in int hint, in int data);
+  oneway void mtkPowerHint(in int hint, in int data);
+  int querySysInfo(in int cmd, in int param);
+  int setMtkPowerCallback(in vendor.mediatek.hardware.mtkpower.IMtkPowerCallback callback);
+  int setMtkScnUpdateCallback(in int scn, in vendor.mediatek.hardware.mtkpower.IMtkPowerCallback callback);
+  int setSysInfo(in int cmd, in String data);
+  oneway void setSysInfoAsync(in int cmd, in String data);
 }
diff --git a/interfaces/hardware/mtkpower/aidl/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl b/interfaces/hardware/mtkpower/aidl/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
new file mode 100644
index 0000000..9a7a0ac
--- /dev/null
+++ b/interfaces/hardware/mtkpower/aidl/vendor/mediatek/hardware/mtkpower/IMtkPowerCallback.aidl
@@ -0,0 +1,14 @@
+//
+// SPDX-FileCopyrightText: 2026 The LineageOS Project
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package vendor.mediatek.hardware.mtkpower;
+
+@VintfStability
+interface IMtkPowerCallback {
+    oneway void mtkPowerHint(in int hint, in int data);
+    oneway void notifyAppState(
+            in String packageName, in String activityName, in int pid, in int state, in int uid);
+    oneway void notifyScnUpdate(in int scn, in int data);
+}
diff --git a/interfaces/hardware/mtkpower/aidl/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl b/interfaces/hardware/mtkpower/aidl/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
new file mode 100644
index 0000000..1c18265
--- /dev/null
+++ b/interfaces/hardware/mtkpower/aidl/vendor/mediatek/hardware/mtkpower/IMtkPowerService.aidl
@@ -0,0 +1,24 @@
+//
+// SPDX-FileCopyrightText: 2026 The LineageOS Project
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package vendor.mediatek.hardware.mtkpower;
+
+import vendor.mediatek.hardware.mtkpower.IMtkPowerCallback;
+
+@VintfStability
+interface IMtkPowerService {
+    int perfCusLockHint(in int hint, in int duration, in int pid);
+    int perfLockAcquire(
+            in int hdl, in int duration, in int[] boostList, in int pid, in int reserved);
+    oneway void perfLockRelease(in int hdl, in int reserved);
+    int perfLockReleaseSync(in int hdl, in int reserved);
+    oneway void mtkCusPowerHint(in int hint, in int data);
+    oneway void mtkPowerHint(in int hint, in int data);
+    int querySysInfo(in int cmd, in int param);
+    int setMtkPowerCallback(in IMtkPowerCallback callback);
+    int setMtkScnUpdateCallback(in int scn, in IMtkPowerCallback callback);
+    int setSysInfo(in int cmd, in String data);
+    oneway void setSysInfoAsync(in int cmd, in String data);
+}
